pkgs/development/compilers/gcc/patches/default.nix

```diff
@@ -284,6 +298,11 @@ in
   })
 ]
 
+# The clang-based assembler used in darwin.cctools-llvm (LLVM >11) does not support piping input.
+# Fortunately, it does not exhibit the problem GCC has with the cctools assembler.
+# This patch can be dropped should darwin.cctools-llvm ever implement support.
+++ optional (!atLeast7 && hostPlatform.isDarwin && lib.versionAtLeast (lib.getVersion stdenv.cc) "12") ./4.9/darwin-clang-as.patch
```

Comment:
```suggestion
++ optional (!atLeast6 && hostPlatform.isDarwin && lib.versionAtLeast (lib.getVersion stdenv.cc) "12") ./4.9/darwin-clang-as.patch
```

Reply:
(or else move it to the "gcc 6 and earlier" section if you want it applied to `gcc6`)

Reply:
It should apply to GCC 6, so I’ll move it to the correct section.